File: backend/src/Pawora.Core/Entities/Pet.cs
namespace Pawora.Core.Entities;

public class Pet : BaseEntity
{
    public Guid UserId { get; set; }
    public User User { get; set; } = null!;

    public string Name { get; set; } = string.Empty;
    public string Species { get; set; } = string.Empty;
    public string? Breed { get; set; }
    public DateTime? BirthDate { get; set; }
    public string? ImageUrl { get; set; }
}

File: mobile/lib/models/pet.dart
class Pet {
  final String id;
  final String userId;
  final String name;
  final String species;
  final String? breed;
  final String? imageUrl;
  final String? birthDate;

  Pet({
    required this.id,
    required this.userId,
    required this.name,
    required this.species,
    this.breed,
    this.imageUrl,
    this.birthDate,
  });

  factory Pet.fromJson(Map<String, dynamic> json) {
    return Pet(
      id: json['id'] as String,
      userId: json['userId'] as String,
      name: json['name'] as String,
      species: json['species'] as String,
      breed: json['breed'] as String?,
      imageUrl: json['imageUrl'] as String?,
      birthDate: json['birthDate'] as String?,
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'id': id,
      'userId': userId,
      'name': name,
      'species': species,
      'breed': breed,
      'imageUrl': imageUrl,
      'birthDate': birthDate,
    };
  }
}
